crates/matrix-sdk/src/client/mod.rs

```diff
@@ -232,45 +195,25 @@ impl Client {
     /// let alice = UserId::parse("@alice:example.org")?;
     ///
     /// // Now let's try to discover the homeserver and create a client object.
-    /// let client = Client::new_from_user_id(&alice).await?;
+    /// let client = Client::for_user_id(&alice).await?;
     ///
     /// // Finally let's try to login.
     /// client.login(alice, "password", None, None).await?;
     /// # Result::<_, matrix_sdk::Error>::Ok(()) });
     /// ```
     ///
     /// [spec]: https://spec.matrix.org/unstable/client-server-api/#well-known-uri
-    pub async fn new_from_user_id(user_id: &UserId) -> Result<Self> {
-        let config = ClientConfig::new();
-        Client::new_from_user_id_with_config(user_id, config).await
+    pub async fn for_user_id(user_id: &UserId) -> Result<Self, HttpError> {
```

Comment:
This now sounds like the `Client` can be used only for the given user id, this isn't strictly true since you can later on log in using a completely different one.

I would also argue that the value of the constructor is diminished, since you can't set a store. Almost nobody will want to use the `Client` without setting a store, people that would use this method are either:

1. Just playing around before getting serious with the SDK
2. Unaware that this method isn't what they want

I think that just removing it would avoid the confusion.